### PR TITLE
New Package: lycheeverse.lychee 0.10.3

### LINKED ISSUE #91395: [Package Request]:  lychee
### Package Requested

- [X] I would like someone else to build the manifest.
- [ ] I would like help so I can submit the manifest.
- [X] I have performed a search and couldn't find this package.
- [X] I have checked that the installer for this package is not contained in a .zip file
- [X] I think there is a new version available and I have provided the URL.

### Please provide the following information

Publisher: https://github.com/lycheeverse
Publisher Url: https://github.com/lycheeverse/lychee
Package Name: lychee
Description:  ⚡ Fast, async, stream-based link checker written in Rust. Finds broken URLs and mail addresses inside Markdown, HTML, reStructuredText, websites and more! 
Package Version: 0.10.3
Installer URL: https://github.com/lycheeverse/lychee/releases/download/v0.10.3/lychee-v0.10.3-windows-x86_64.exe

Current Package Identifier (for new versions):

### PR DESCRIPTION
**Pull request by Trenly**
* Resolves #91395
* Related to lycheeverse/lychee#59

Has a runtime dependency on VCRedist2015+

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/91422)

### CODE REVIEW

**Comment by Trenly** [comment]
@msftbot dependencies

**Comment by Trenly** [comment]
@msftbot reset feedback

File: manifests/l/lycheeverse/lychee/0.10.3/lycheeverse.lychee.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+Commands:
+  - lychee
+ReleaseDate: 2022-11-08
+Installers:
+- Architecture: x86
+  InstallerUrl: https://github.com/lycheeverse/lychee/releases/download/v0.10.3/lychee-v0.10.3-windows-x86_64.exe
+  InstallerSha256: A75CDAC942DBE9AF1FAAD2A494C748EB19CC365541C77329398692448CA0BC97
+  Dependencies:
+    PackageDependencies:
+      - PackageIdentifier: Microsoft.VCRedist.2015+.x86
```
Comment (stephengillie) [inline manifests/l/lycheeverse/lychee/0.10.3/lycheeverse.lychee.installer.yaml:17]:
```suggestion
```

Comment (Trenly) [reply manifests/l/lycheeverse/lychee/0.10.3/lycheeverse.lychee.installer.yaml:17]: Does it not work on x86? Oh well ¯\\\_(ツ)_/¯

Comment (stephengillie) [reply manifests/l/lycheeverse/lychee/0.10.3/lycheeverse.lychee.installer.yaml]: Based on the x86 validation error, `The specified executable is not a valid application for this OS platform.`, I believe the application may be only x64.

**Review by stephengillie — APPROVED** [review]

**Pull request merged**